compactor2/src/components/ir_planner/planner_v1.rs

```diff
@@ -228,13 +230,15 @@ impl IRPlanner for V1IRPlanner {
                 PlanIR::Compact {
                     files,
                     target_level,
+                    reason,
                 }
             } else {
                 // split compact query plan to split the result into multiple files
                 PlanIR::Split {
                     files,
                     split_times,
                     target_level,
+                    reason: SplitReason::TriedToCompactTooMuch(reason),
```

Comment:
This is definitely an interesting case that making three `Reason` types (one to go with each of `Split`, `Compact`, `None`) allowed me to discover: this is in the `compact_plan` function but it creates a split plan??? Is this intentional?

Reply:
I think the problem is that "compact" is used to mean two different things in different places.

Compactor algorithm:
1. "compact" is a decision to "try and combine all the files together"
2. "split" means "explicit split on time ranges"

However to keep to the target output size, a "compact" may need to make multiple output files

PlanIR / reorg panner:
1. "compact" means really make a single output file
2. "split" means split at times


So I think it wold be helpfu to use different terms in one of those places (maybe "combine" instead of "compact"?)

Reply:
> So I think it wold be helpfu to use different terms in one of those places (maybe "combine" instead of "compact"?)

I'm into that! Which place should be "combine" though?

Reply:
I think we should define terms here:

1. Regarding out algorithms: we have to decide whether to `compact` & `split`. This PR is aim to distinguish that and show it in the layout tests for us to know the exact steps the compactor decides for each test.
2. When we do the `compact` above, we have to split if the output is large as @alamb explains about. So technically when you come inside this function, the reason here must be `compact`

I agree we should have different words for them but let us do it in a separate PR

Reply:
So you can either 
```suggestion
                    reason,
```

Or add some extra information for us to know this is a one of the file created by the **compact**'s `reason`

Reply:
> So I can't take that suggestion because this reason field on the PlanIR::Split is of type SplitReason, and the reason coming in to this compact_plan function is a CompactReason. I don't think it's a good idea to combine the SplitReason and CompactReason types; I think keeping them separate will prevent problems where split/compact decisions could get tangled up.

If so, lat make it something:  `SplitReason::CompactAndSplitOutput`

>  So what extra information would you like added here, exactly?
No extra information needed if we go with `SplitReason::CompactAndSplitOutput`